### PR TITLE
Add bugsnag to kurl.sh

### PR DESCRIPTION
**Pull request by jala-dx**
#### What this PR does / why we need it:
Adds bugsnag to kurl.sh

#### Which issue(s) this PR fixes:
https://app.shortcut.com/replicated/story/57005/add-bugsnag-to-kurl-sh

#### Special notes for your reviewer:
Please refer to the above linked story.

```

- kurl.sh is not a typical React app, instead it is based on Gatsby )there is no ReactDOM.render anywhere in the code. )

- Following this guide https://www.benmvp.com/blog/integrating-bugsnag-into-gatsbyjs-netlify/#gatsbyjs
   Moved the entire bugsnag integration to gatsby-browser.js
 
- Gatsby does however, have a wrapRootElement() hook that allows us to add Bugsnag’s ErrorBoundary before Gatsby takes 
   over to continue doing its thing. We call it in gatsby-browser.js.

- It can also be modified at index.js (at the root level). There is no change in the backtrace/breadcrumbs in the bugsnag dashboard regardless of which method.

- Verified if the bugsnag notifications reaches the Dashboard.  
https://app.bugsnag.com/replicated-inc/kurl-dot-sh/overview?release_stage=development


```

#### Does this PR introduce a user-facing change?
No

#### Does this PR require documentation?
No

### CODE REVIEW

File: .env.development [hunk 1]
```diff
@@ -5,4 +5,5 @@ CLUSTER_API_URL="cluster.kurl.sh"
 KURL_URL="https://localhost:8072"
 ADDONS_JSON="https://kurl-sh.s3.amazonaws.com/staging/addons-gen.json"
 SUPPORTED_VERSIONS_JSON="https://kurl-sh.s3.amazonaws.com/staging/supported-versions-gen.json"
-IS_DEVELOPMENT="true"
\ No newline at end of file
+IS_DEVELOPMENT="true"
+BUGSNAG_API_KEY="d9853b1ecad8ac8750308fe7ce9335b8"
```
Comment (jala-dx) [inline .env.development:9]: github is odd?, there is no newline between 8 and 9.

File: gatsby-browser.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+const React = require('react')
+const Bugsnag = require('@bugsnag/js').default
+const BugsnagPluginReact = require('@bugsnag/plugin-react').default
+
+Bugsnag.start({
+  apiKey: 'd9853b1ecad8ac8750308fe7ce9335b8',
```
Comment (jala-dx) [inline gatsby-browser.js:6]: need to fix this.

**Review by emosbaugh — CHANGES_REQUESTED** [review]
I think you need to set the release stage unless this is somehow set by gatsby automatically https://docs.bugsnag.com/platforms/go/other/configuration-options/#releasestage

**Review by emosbaugh — APPROVED** [review]

**Comment by jala-dx** [comment]
> I think you need to set the release stage unless this is somehow set by gatsby automatically https://docs.bugsnag.com/platforms/go/other/configuration-options/#releasestage

yes you are right. We need to set the `release stages` and also `notifyReleaseStages` so that the notifications actually reach on  prod and staging.

**Pull request merged**